core/vm/contracts.go

```diff
@@ -180,10 +180,12 @@ func init() {
 
 func activePrecompiledContracts(rules params.Rules) PrecompiledContracts {
 	switch {
+	case rules.IsDia:
+		return PrecompiledContractsStartingFromArbOS50
 	case rules.IsStylus:
-		return PrecompiledContractsArbOS30
+		return PrecompiledContractsAfterArbOS30
```

Comment:
How about changing this too?
```suggestion
		return PrecompiledContractsStartingFromArbOS30
```